### PR TITLE
python-numpy: fix for armv5tel-musl.

### PR DESCRIPTION
**Pull request by fosslinux**
in fenv.h, musl disables the normal set of constants that you would get
on arm with an arm device with an fpu (armv6 and up), but armv5tel on
void uses softfloat all the time. But python3-numpy dosen't use these to
interact with the fpu (the compiler would generate those instructions)
and rather just uses those constants for a generic GCC implementation.
So we give them out anyway.

python-numpy uses the same patch from python3-numpy.

### CODE REVIEW

**Comment by fosslinux** [comment]
Just realized that python3-numpy would have the same issue, restructured a bit

File: srcpkgs/python-numpy/template [hunk 1]
```diff
@@ -15,6 +14,17 @@ distfiles="https://github.com/numpy/numpy/archive/v${version}.tar.gz"
 checksum=3c82a9b8616e3096a79a2af9c288d8ed4013a10fc7baf3eaf54655309734dadd
 alternatives="numpy:f2py:/usr/bin/f2py2"
 
+post_patch() {
+	case "${XBPS_TARGET_MACHINE}" in
+		armv5tel-musl)
+			cp "${XBPS_SRCPKGDIR}/python3-numpy/files/fenv-constants.h" \
```
Comment (sgn) [inline srcpkgs/python-numpy/template:20]: `$FILESDIR`

Comment (fosslinux) [reply srcpkgs/python-numpy/template:20]: No, `python3-numpy` is seperate to `python-numpy`.

**Comment by fosslinux** [comment]
ping

File: srcpkgs/python-numpy/template (hunk 1 above)
Comment (ericonr) [reply srcpkgs/python-numpy/template:20]: Could it be symlinked instead?

Comment (fosslinux) [reply srcpkgs/python-numpy/template:20]: i guess

Comment (ericonr) [reply srcpkgs/python-numpy/template]: I was thinking more of symlinking the entire folder D:

Comment (fosslinux) [reply srcpkgs/python-numpy/template]: I don't really like that.. the `files/` folder should be independent of any one specific patch/fix that is applied.

**Comment by sgn** [comment]
I think the fix should go to `musl`?

**Comment by sgn** [comment]
It's weird that `musl-devel` for armv5tel-musl has those macro defined in `/usr/include/bits/fenv.h`

**Comment by ericonr** [comment]
@sgn LGTM, if you've cleared up everything on your end.

File: srcpkgs/python-numpy/template (hunk 1 above)
Comment (ahesford) [reply srcpkgs/python-numpy/template]: I think both of these options are bad. If somebody changes or removes the `python3-numpy` files (which becomes more likely the further the versions of the py2 and py3 packages diverge), the symlink or the cross-template copy will break. If this package needs this header, let it own a copy.

Comment (fosslinux) [reply srcpkgs/python-numpy/template]: I agree, I only changed it because @ericonr asked for it, but I can change it back.

Comment (fosslinux) [reply srcpkgs/python-numpy/template]: I've moved it back.

**Comment by fosslinux** [comment]
Fixing tests is outside the scope of this PR.

**Comment by ahesford** [comment]
Looks like you leaked an `m4` patch change into one of the commits.

**Comment by fosslinux** [comment]
Oops

**Comment by ahesford** [comment]
I have combined this with a `python3-numpy` update and will merge soon.

**Pull request closed without merging**